gsadjust/GSadjust.py

```diff
@@ -503,11 +503,15 @@ def open_file_dialog(self, open_type):
 
         """
         fname, _ = QtWidgets.QFileDialog.getOpenFileName(
-            caption='Open file', directory=self.settings.value('current_dir')
+            caption='Open file',
+            directory=self.settings.value('current_dir'),
         )
 
         if fname:
             self.settings.setValue('current_dir', os.path.dirname(fname))
+
+            # FIXME: This check wouldn't be neccessary if we restrict the file extensions in the open dialog.
```

Comment:
We can avoid these kinds of checks by limiting the filenames the open/save dialogs can produce. By adding specific filters, only matching filenames are allowed.